kratos/utilities/reduction_utilities.h

```diff
@@ -278,6 +278,36 @@ class AccumReduction
     }
 };
 
+template<class MapType>
+class MapReduction
```

Comment:
Please check with what I added in #10339, I would think that `merge` is faster

In any case, can you then please also update the custom solution?

Reply:
done :D